### ISSUE TITLE
Send a completely different welcome mail

### ISSUE THREAD

**Issue by drSun** [2amigos/yii2-usuario#367]
Hello!
Is there any way not to send a default welcome email and send my own email in overrode RegisterController? Can't find an option. In fact, I only need to change a layout and html of welcome email, but without touching other emails (recovery etc).

Thank you.

**Comment by maxxer** [comment]
Did you try [overriding views](https://yii2-usuario.readthedocs.io/en/latest/enhancing-and-overriding/overriding-views/)?

**Comment by drSun** [comment]
> Did you try [overriding views](https://yii2-usuario.readthedocs.io/en/latest/enhancing-and-overriding/overriding-views/)?

Yes, thank you, I know about this possibility, but then I will override all other emails as well, what is not needed. But if this is the only option, I would use it, of course.

**Comment by maxxer** [comment]
No, you can just override the ones you need. The not found ones will be fetched from the original source

**Comment by drSun** [comment]
> No, you can just override the ones you need. The not found ones will be fetched from the original source

I understand this, but I'm talking about layouts, not only about emails :) can I use a different layout only for the welcome email?

**Comment by drSun** [comment]
Really, I'd rather switch off sending default welcome email, and send my own using event 'after registration'. Because as I see, there is even no opportunity to embed images in MailFactory, is it?

**Comment by maxxer** [comment]
https://yii2-usuario.readthedocs.io/en/latest/installation/configuration-options/#enableemailconfirmation-type-boolean-default-true

**Comment by drSun** [comment]
A very useful option, but it only regulates including or not a confirmation link in welcome mail subject.

Anyway, thank you for your help. I decided to copy-paste RegistrationController->actionRegister() and override a MailFactory method makeWelcomeMailerService(). A bad coding style, but can imagine nothing better :(

**Comment by edegaudenzi** [comment]
dead link
> [https://yii2-usuario.readthedocs.io/en/...](https://yii2-usuario.readthedocs.io/en/latest/installation/configuration-options/#enableemailconfirmation-type-boolean-default-true)

Working link: https://yii2-usuario.readthedocs.io/en/latest/install/configuration-options/#enableemailconfirmation-type-boolean-default-true
